mempool/v1/mempool_test.go

```diff
@@ -209,38 +212,68 @@ func TestTxMempool_ReapMaxBytesMaxGas(t *testing.T) {
 		return priorities[i] > priorities[j]
 	})
 
-	ensurePrioritized := func(reapedTxs types.Txs) {
+	ensurePrioritized := func(reapedTxs types.Txs, testCase string) {
 		reapedPriorities := make([]int64, len(reapedTxs))
 		for i, rTx := range reapedTxs {
 			reapedPriorities[i] = txMap[rTx.Key()].priority
 		}
 
-		require.Equal(t, priorities[:len(reapedPriorities)], reapedPriorities)
+		require.Equal(t, priorities[:len(reapedPriorities)], reapedPriorities, testCase)
 	}
-
+	t.Run("ReapMaxBytesMaxGas", func(t *testing.T) {
+		testReapMaxBytesMaxGas(t, txmp, tTxs, expectedSize, ensurePrioritized)
+	})
+	t.Run("ReapMaxTxs", func(t *testing.T) {
+		testReapMaxTxs(t, txmp, tTxs, expectedSize, ensurePrioritized)
+	})
+}
+func testReapMaxBytesMaxGas(t *testing.T, txmp *TxMempool, tTxs []testTx, expectedSize int64, checkFunc func(reapedTxs types.Txs, testCase string)) {
 	// reap by gas capacity only
 	reapedTxs := txmp.ReapMaxBytesMaxGas(-1, 50)
-	ensurePrioritized(reapedTxs)
+	checkFunc(reapedTxs, fmt.Sprint(t.Name(), "reap by gas capacity only"))
 	require.Equal(t, len(tTxs), txmp.Size())
-	require.Equal(t, int64(5690), txmp.SizeBytes())
+	require.Equal(t, expectedSize, txmp.SizeBytes())
 	require.Len(t, reapedTxs, 50)
 
 	// reap by transaction bytes only
-	reapedTxs = txmp.ReapMaxBytesMaxGas(1000, -1)
-	ensurePrioritized(reapedTxs)
+	reapedTxs = txmp.ReapMaxBytesMaxGas(1500, -1)
```

Comment:
This was ultimately the fix, going from 1000 to 1500.

Does anyone know if this is an acceptable way to fix the test? I couldn't understand where the value of 1000 came from.

Reply:
@tzdybal do you know the history of this number?

Reply:
1000 comes from the https://github.com/rollkit/rollkit/blob/v0.10.x/mempool/v1/mempool.go#L344, where the test is making sure that at least 16 txs gets reaped with tx bytes (56 or 57). note that, the function computes the proto converted size, so its slightly larger.